gateway-service-impl/src/test/java/org/hypertrace/gateway/service/common/datafetcher/QueryServiceEntityFetcherTests.java

```diff
@@ -81,6 +91,167 @@ public void setup() {
+
+    TimeAggregation timeAggregation1 =
+        buildTimeAggregation(
+            30, API_NUM_CALLS_ATTR, FunctionType.SUM, "SUM_API.numCalls", List.of());
+
+    TimeAggregation timeAggregation2 =
+        buildTimeAggregation(
+            30,
+            API_NUM_CALLS_ATTR,
+            FunctionType.AVGRATE,
+            "AVGRATE_API.numCalls",
+            List.of(
+                Expression.newBuilder()
+                    .setLiteral(
+                        LiteralConstant.newBuilder()
+                            .setValue(Value.newBuilder().setLong(60).setValueType(ValueType.LONG)))
+                    .build()));
+
+    EntitiesRequest entitiesRequest =
+        EntitiesRequest.newBuilder()
+            .setEntityType(entityType.name())
+            .setStartTimeMillis(startTime)
+            .setEndTimeMillis(endTime)
+            .addTimeAggregation(timeAggregation1)
+            .addTimeAggregation(timeAggregation2)
+            .setFilter(
+                Filter.newBuilder()
+                    .setOperator(AND)
+                    .addChildFilter(
+                        EntitiesRequestAndResponseUtils.getTimeRangeFilter(
+                            "API.startTime", startTime, endTime))
+                    .addChildFilter(generateEQFilter(API_DISCOVERY_STATE_ATTR, "DISCOVERED")))
+            .addAllOrderBy(orderByExpressions)
+            .setLimit(limit)
+            .setOffset(offset)
+            .build();
+
+    EntitiesRequestContext entitiesRequestContext =
+        new EntitiesRequestContext(
+            tenantId, startTime, endTime, entityType.name(), "API.startTime", requestHeaders);
+
+    QueryRequest expectedQueryRequest =
+        QueryRequest.newBuilder()
+            .addSelection(
+                QueryAndGatewayDtoConverter.convertToQueryExpression(
+                    timeAggregation1.getAggregation()))
+            .addSelection(
+                QueryAndGatewayDtoConverter.convertToQueryExpression(
+                    timeAggregation2.getAggregation()))
+            .setFilter(
+                createQsRequestFilter(
+                    API_START_TIME_ATTR,
+                    API_ID_ATTR,
+                    startTime,
+                    endTime,
+                    createStringFilter(API_DISCOVERY_STATE_ATTR, Operator.EQ, "DISCOVERED")))
+            .addAllGroupBy(
+                List.of(API_ID_ATTR).stream()
+                    .map(QueryRequestUtil::createColumnExpression)
+                    .collect(Collectors.toList()))
+            .addGroupBy(createTimeColumnGroupByExpression(API_START_TIME_ATTR, 30))
+            .setOffset(0)
+            .setLimit(10000)
+            .build();
+
+    List<ResultSetChunk> resultSetChunks =
+        List.of(
+            getResultSetChunk(
+                List.of("API.apiId", "timeColumn", "SUM_API.numCalls", "AVGRATE_API.numCalls"),
+                new String[][] {
+                  {"apiId1", "10000", "34", "43"},
```

Comment:
The last two columns will be the same till here  (unless the time range is different) as `AVGRATE_API` internally asking SUM from Pinot handler.
```
{"apiId1", "10000", "34", "34"},
{"apiId2", "20000", "34", "34"}
```